geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -2070,11 +2163,12 @@ public void close(String reason, Throwable systemFailureCause, boolean keepAlive
     final boolean isDebugEnabled = logger.isDebugEnabled();
 
     synchronized (GemFireCacheImpl.class) {
-      // fix for bug 36512 "GemFireCache.close is not thread safe"
-      // ALL CODE FOR CLOSE SHOULD NOW BE UNDER STATIC SYNCHRONIZATION
-      // OF synchronized (GemFireCache.class) {
+      // ALL CODE FOR CLOSE SHOULD NOW BE UNDER STATIC SYNCHRONIZATION OF GemFireCacheImpl.class
       // static synchronization is necessary due to static resources
       if (isClosed()) {
+        if (!skipAwait) {
```

Comment:
This changes the existing behavior...Where subsequent close calls are returned immediately...I guess its ok; as the first close can be any thread depending on the scheduling.